service/history/replication/task_ack_manager.go

```diff
@@ -233,6 +238,7 @@ TaskInfoLoop:
 		ReplicationTasks:       replicationTasks,
 		HasMore:                hasMore,
 		LastRetrievedMessageID: readLevel,
+		Size:                   taskSize,
```

Comment:
I believe the whole size calculation could be done more universally within a middleware. It already knows serialized content in bytes. Then is become the question how to use it. One way I can think - is to set a value in the passed context. I will try to come up with some PR for that.

Reply:
Please check whether this could be suitable: https://github.com/uber/cadence/pull/4522

It is not exactly the same - it contains serialized message size, but after gathering all shards it would get serialized again, so it is more or less the same. I guess we could also reserve some additional buffer, for the margin of error. Lets say, stop accumulating tasks once we pass 90% of max message size.

Reply:
Thank you. Yes. This makes more sense

Reply:
I will see how can I get the response info from the context